src/main/java/com/uid2/shared/auth/AuthorizableStore.java

```diff
@@ -0,0 +1,78 @@
+package com.uid2.shared.auth;
+
+import com.uid2.shared.secret.KeyHasher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+
+public class AuthorizableStore<T extends IAuthorizable> {
+    private class AuthorizableStoreSnapshot {
```

Comment:
We can make this static as the nested class doesn't reference anything from the outer class

Reply:
I just tried it, but it complained it couldn't access the generic type I think

Reply:
Ah right... because it does reference something with the other class, which is the `T`